### PR TITLE
audit: 2.8.5 -> 3.1

### PR DESCRIPTION
**Pull request by henrirosten**
###### Description of changes

Update audit to current latest version, see: https://github.com/linux-audit/audit-userspace/releases/tag/v3.1.

This change was motivated by manual test run of https://github.com/tiiuae/sbomnix/tree/main/scripts/nixupdate, which pointed out that audit is currently not up-to-date in nix unstable.

Tested locally that the following pass on x86_64-linux:
- `nix-build -A audit`
- `nix-build -A pkgsStatic.audit`
- `nix-build -A pkgsCross.aarch64-multiplatform.pkgsStatic.audit`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by emilylange — APPROVED** [review]
Thanks for picking this up!

I am currently packaging and writing a module for https://github.com/threathunters-io/laurel and were surprised to see that NixOS still uses [v2.8.5](https://github.com/linux-audit/audit-userspace/releases/tag/v2.8.5) from 2019

File: pkgs/os-specific/linux/audit/default.nix [hunk 1]
```diff
@@ -37,40 +37,9 @@ stdenv.mkDerivation rec {
...
-  #         https://github.com/linux-audit/audit-userspace/pull/25
-  #       is available with the next release.
   patches = [
-    ./patches/weak-symbols.patch
-    (fetchpatch {
-      # upstream build fix against -fno-common compilers like >=gcc-10
-      url = "https://github.com/linux-audit/audit-userspace/commit/017e6c6ab95df55f34e339d2139def83e5dada1f.patch";
-      sha256 = "100xa1rzkv0mvhjbfgpfm72f7c4p68syflvgc3xm6pxgrqqmfq8h";
-    })
+    ./patches/fix-static.patch
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/audit/default.nix:41]:
```suggestion
    ./fix-static.patch
```

Comment (henrirosten) [reply pkgs/os-specific/linux/audit/default.nix]: Fixed in: 50c40a8f6ab80600cf095855ecd7aa972cb6eac3

**Comment by RaitoBezarius** [comment]
@ofborg eval

**Pull request merged**